view_conversations/conv-viewer-webapi/src/db_types.ts

```diff
@@ -0,0 +1,172 @@
+  public abstract fromSpannerResultField(row:spanner.ResultField) : T | null;
+}
+
+// TODO(ldixon): add additional constraints, e.g. regexp matching.
+class StringFieldHandler extends SpannerFieldHandler<string> {
+  public fromSpannerResultField(field:spanner.ResultField) : string | null {
+    if(!(field === null || typeof(field) === 'string')) {
+      console.error('Error: field:');
+      console.dir(field);
+      throw Error(`For ${this.fieldName}: expected field:string, not: ${typeof(field)}`);
+    }
+    return field;
+  }
+}
+
+class BytesFieldHandler extends SpannerFieldHandler<string> {
+  public fromSpannerResultField(field:spanner.ResultField) : string | null {
+    if (field === null){
+      return null;
+    }
+    if(!(field instanceof Uint8Array)) {
+        console.error('Error: field:');
+        console.dir(field);
+        throw Error(`For ${this.fieldName}: expected field: byte array, not: ${typeof(field)}`);
+    }
+    return ab2str(field as Uint8Array);
+  }
+}
+
+class ArrayFieldHandler extends SpannerFieldHandler<string[]> {
+  public fromSpannerResultField(field:spanner.ResultField) : string[] | null {
+    if (field === null) {
+      return field;
+    }
+    if (!(field instanceof Array)) {
+      console.error('Error: field:');
+      console.dir(field);
+      throw Error(`For ${this.fieldName}: expected field: string array, not: ${typeof(field)}`);
+    }
+    return field as string[];
+  }
+}
+
+
+
+class IntFieldHandler extends SpannerFieldHandler<number> {
+  public fromSpannerResultField(field:spanner.ResultField) : number | null {
+    if(typeof(field) === 'string') {
+      throw Error(`For ${this.fieldName}: expected field.value:string, but field:string`);
+    } else if (field instanceof Date) {
+      throw Error(`For ${this.fieldName}: expected field.value:string, but field:Date`);
+    } else if (field === null) {
+      return null;
+    }
+    return parseInt((field as spanner.ResultValue).value, 10);
+  }
+}
+
+class FloatFieldHandler extends SpannerFieldHandler<number> {
+  public fromSpannerResultField(field:spanner.ResultField) : number | null {
+    if(typeof(field) === 'string') {
+      throw Error(`For ${this.fieldName}: expected field.value:string, but field:string`);
+    } else if (field instanceof Date) {
+      throw Error(`For ${this.fieldName}: expected field.value:string, but field:Date`);
+    } else if (field === null) {
+      return null;
+    }
+    return parseFloat((field as spanner.ResultValue).value);
+  }
+}
+
+class TimestampFieldHandler extends SpannerFieldHandler<Date> {
+  public fromSpannerResultField(field:spanner.ResultField) : Date | null {
+    if (!(field === null || field instanceof Date)) {
+      throw Error(`For ${this.fieldName}: expected field:Date, but field:${typeof(field)}`);
+    }
+    return field;
+  }
+}
+
+interface HandlerSet { [fieldName:string] : SpannerFieldHandler<any>; };
```

Comment:
Avoid using the type any. Instead use the actual types this could be, which looks like Date|number|string|string[].